gtsam/base/Lie.h

```diff
@@ -301,22 +338,21 @@ T BCH(const T& X, const T& Y) {
   return T(X + Y + _2 * X_Y + _12 * bracket(X - Y, X_Y) - _24 * bracket(Y, bracket(X, X_Y)));
 }
 
-/**
- * Declaration of wedge (see Murray94book) used to convert
- * from n exponential coordinates to n*n element of the Lie algebra
- */
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V43
+/// @deprecated: use T::Hat
 template <class T> Matrix wedge(const Vector& x);
+#endif
 
 /**
  * Exponential map given exponential coordinates
- * class T needs a wedge<> function and a constructor from Matrix
+ * class T needs a constructor from Matrix.
  * @param x exponential coordinates, vector of size n
  * @ return a T
  */
 template <class T>
-T expm(const Vector& x, int K=7) {
-  Matrix xhat = wedge<T>(x);
-  return T(expm(xhat,K));
+T expm(const Vector& x, int K = 7) {
+  auto xhat = T::Hat(x);
```

Comment:
I prefer statically defined types for code. I can update this in another PR.

Reply:
It’s already fixed. In this case bTW Hat is defined differently for every T so auto makes sense, except it will need to be cast anyway to a Matrix for expm